lib/asset_sync/storage.rb

```diff
@@ -202,6 +202,10 @@ def upload_file(f)
         })
       end
 
+      if config.azure_rm?
+        file[:content_type] = file[:content_type].content_type if file[:content_type].is_a?(::MIME::Type)
```

Comment:
Don't quite understand why this is needed

Reply:
I'm not sure, sorry...
but, I got an error as below without the content_type conversion.

```
rake aborted!
NoMethodError: undefined method `length' for #<MIME::Type:0x007fda42e79040>
vendor/bundle/ruby/2.3.0/gems/azure-storage-0.11.5.preview/lib/azure/storage/core/utility.rb:112:in `parse_charset_from_content_type'
vendor/bundle/ruby/2.3.0/gems/azure-storage-0.11.5.preview/lib/azure/storage/blob/blob_service.rb:54:in `call'
vendor/bundle/ruby/2.3.0/gems/azure-storage-0.11.5.preview/lib/azure/storage/blob/block.rb:114:in `create_block_blob'
vendor/bundle/ruby/2.3.0/gems/fog-azure-rm-0.3.8/lib/fog/azurerm/requests/storage/create_block_blob.rb:31:in `create_block_blob'
vendor/bundle/ruby/2.3.0/gems/fog-azure-rm-0.3.8/lib/fog/azurerm/models/storage/file.rb:250:in `save_blob'
vendor/bundle/ruby/2.3.0/gems/fog-azure-rm-0.3.8/lib/fog/azurerm/models/storage/file.rb:93:in `save'
vendor/bundle/ruby/2.3.0/gems/fog-core-1.45.0/lib/fog/core/collection.rb:50:in `create'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/asset_sync/storage.rb:205:in `upload_file'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/asset_sync/storage.rb:220:in `block in upload_files'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/asset_sync/storage.rb:218:in `each'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/asset_sync/storage.rb:218:in `upload_files'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/asset_sync/storage.rb:234:in `sync'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/asset_sync/asset_sync.rb:29:in `block in sync'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/asset_sync/asset_sync.rb:51:in `with_config'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/asset_sync/asset_sync.rb:28:in `sync'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/tasks/asset_sync.rake:5:in `block (2 levels) in <top (required)>'
vendor/bundle/ruby/2.3.0/bundler/gems/asset_sync-2fa4d474d97e/lib/tasks/asset_sync.rake:28:in `block in <top (required)>'
```

I think `Azure::Storage::Core::Utility#parse_charset_from_content_type` needs `content_type` as a String, not as a MIME::Type, because it calls `length` and `split(';')` methods of `content_type`. so, I put the code to convert from MIME::Type to String.

```ruby
module Azure::Storage
  module Core
    module Utility
      ...
      def parse_charset_from_content_type(content_type)
        if (content_type && content_type.length > 0)
          charset = content_type.split(';').delete_if { |attribute| !attribute.lstrip.start_with?('charset=') }.map { |x| x.lstrip }[0]
          charset['charset='.length...charset.length] if charset
        end
      end
```

Reply:
I guess add a code comment to briefly explain this

Reply:
I added a comment!